packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -309,7 +345,27 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
 
         if (isDynamicImport && insertPreload) {
           needPreloadHelper = true
-          str().prependLeft(expStart, `${preloadMethod}(() => `)
+          const { declaration, names, chains } = dynamicImports[expEnd] || {}
+          if (names) {
+            str().prependLeft(
+              expStart,
+              `${preloadMethod}(async () => { ${declaration} = await `,
+            )
+            str().appendRight(expEnd, `;return ${names}}`)
+          } else if (chains) {
+            const name = chains.match(/\.([^.?]+)/)?.[1] || ''
+            str().prependLeft(
+              expStart,
+              `${preloadMethod}(async () => { const __vite_temp__ = (await `,
+            )
+            str().appendRight(
+              expEnd,
+              `).${name}; return { ${name}: __vite_temp__ }}`,
+            )
+          } else {
+            str().prependLeft(expStart, `${preloadMethod}(() => `)
+          }
```

Comment:
Would be great to have your explanation for how this would look like here too before & after transform here too.